### PR TITLE
feat: introduced a new MultiEntityPicker field 

### PR DESCRIPTION
**Pull request by mrooding**
## Hey, I just made a Pull Request!

As discussed in #16334 this PR adds a new `MultiEntityPicker` scaffolder field.

image
image

It's essentially a clone of the `EntityPicker` with the required modifications to support multiple entities. 

Tests are still lacking, I'd prefer to align first on the approach before adding these.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by alper98** [comment]
Would it be possible to support showing the title of the entity chosen?

(I am not a maintainer)

**Comment by mrooding** [comment]
definitely not stale. I'd love feedback from @rugvip or @camilaibs to get it moving.

@alper98 that should be doable. I'm not sure why the regular `EntityPicker` does not support this but I don't mind adding it behind a flag

**Comment by mrooding** [comment]
Not stale at all. Still waiting for feedback

**Review by Rugvip — CHANGES_REQUESTED** [review]
Sorry about the delay on this one, but it all looks good, thank you! :+1: :tada: 

One request before merge though, could you add a few tests to make sure it keeps working in the future?

**Comment by mrooding** [comment]
@Rugvip definitely, I added them in the latest commit

@alper98 The `title` is not an attribute set on all entities. For my use case, picking multiple users, an option to display the title instead wouldn't help much. I was thinking that maybe it would make more sense to optionally specify a path that would be used for the label. For example, `metadata.title` would allow you to display the title. What do you think?

**Comment by alper98** [comment]
> @Rugvip definitely, I added them in the latest commit
> 
> @alper98 The `title` is not an attribute set on all entities. For my use case, picking multiple users, an option to display the title instead wouldn't help much. I was thinking that maybe it would make more sense to optionally specify a path that would be used for the label. For example, `metadata.title` would allow you to display the title. What do you think?

sounds good to me :) However, might be worth including the maintainers in all this. Perhaps this https://github.com/backstage/backstage/pull/17474 would be relevant to consider?

**Comment by alper98** [comment]
> @Rugvip definitely, I added them in the latest commit
> 
> 
> 
> @alper98 The `title` is not an attribute set on all entities. For my use case, picking multiple users, an option to display the title instead wouldn't help much. I was thinking that maybe it would make more sense to optionally specify a path that would be used for the label. For example, `metadata.title` would allow you to display the title. What do you think?

Sounds goood

**Comment by mrooding** [comment]
not stale

**Review by Rugvip — CHANGES_REQUESTED** [review]
Please make sure all commits are signed with DCO so that we can continue the review 🙏

**Comment by mrooding** [comment]
Hi @Rugvip, all commits are now signed.

**Comment by mrooding** [comment]
Not stale

**Comment by mrooding** [comment]
@Rugvip is there anything left that I can do? Any chance to get this into 1.20? 🙏

File: .changeset/metal-elephants-sit.md
Comment (benjdlambert) [inline .changeset/metal-elephants-sit.md:5]:
```suggestion
Introduced a new `MultiEntityPicker` field that supports selecting multiple Entities
```

File: plugins/scaffolder/src/plugin.tsx
Comment (benjdlambert) [inline plugins/scaffolder/src/plugin.tsx:147]: I think you'll also need to export this from the main `src/index.ts` for people to use. Looks like it might be missing.

You'll then also want to regenerate the api reports, which you can do by doing `yarn build:api-reports` and then commit the changes in markdown.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
This is looking pretty good so far, thanks for all the work so far!

Just 2 small things, need to generate api-reports after we've exported and the changeset.

**Comment by mrooding** [comment]
Hi @benjdlambert 

I've added the export and updated the api report.

File: plugins/scaffolder/src/components/fields/MultiEntityPicker/MultiEntityPicker.tsx
Comment (benjdlambert) [inline plugins/scaffolder/src/components/fields/MultiEntityPicker/MultiEntityPicker.tsx:37]: Think this should be imported from `@rjsf/utils`? Looks like we might need to update the docs now.

**Review by benjdlambert — COMMENTED** [review]
The build seems to be failing, left a suggested fix.

**Comment by mrooding** [comment]
Hey @benjdlambert sorry I missed your reply. I just pushed the change. Could you please re-open the PR? 🙏

**Comment by benjdlambert** [comment]
@mrooding I don't think I can open the PR again, could you create a new PR? It's all greyed out for me :(

**Comment by benjdlambert** [comment]
Screenshot 2023-12-14 at 20 06 55

**Pull request closed without merging**
